src/from_file.py

```diff
@@ -43,7 +43,7 @@ def write_to_version_file(filename, versions):
     contents = json.dumps(versions, sort_keys=True,
                           indent=1, separators=(",", ": "))
     with open(filename, "w") as f:
-        f.write(SHORT_VERSION_PY % contents)
+        f.write(SHORT_VERSION_PY.format(contents))
 
-    print("set %s to '%s'" % (filename, versions["version"]))
+    print("set {} to '{}'".format(filename, versions["version"]))
```

Comment:
Might as well use an f-string if we're here.

```suggestion
    print(f"set {filename} to {versions['version']:r}")
```

Reply:
But then we're modifying functionality, by changing `'%s'` to `{...}` instead of `'{...}'`, aren't we?

Reply:
Also, I suspect you meant `!r` instead of `:r`. It will call [`repr()`](https://docs.python.org/3/library/functions.html#repr), but then how is that helpful?
```python
>>> i = 2
>>> 
>>> f"We print 'i' as '{i}'"
"We print 'i' as '2'"
>>> 
>>> f"We print 'i' as {i!r}"
"We print 'i' as 2"
>>> 
```
We might use triple quotes instead:
```python
f"""set {filename} to '{versions["version"]}'"""
```
But then it gets unreadable.